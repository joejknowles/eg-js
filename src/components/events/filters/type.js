import React from 'react';

const typeOptions = [
  'all',
  'gig',
  'other'
]

export const TypeFilter = (props) => (
  <select onChange={ event => props.onChange(event.target.value) } >
    <option key='all' value=''>Select type</option>
    { typeOptions.map(
      t => <option
        key={ t }
        value={ t }>
        { t }</option>
    ) }
  </select>
);
