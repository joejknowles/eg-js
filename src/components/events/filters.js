import React from 'react';

export const Filters = (props) => (
  <div className='Filters'>
    <input onChange={ event => props.onTypeFilterChange(event.target.value) } ></input>
  </div>
);
