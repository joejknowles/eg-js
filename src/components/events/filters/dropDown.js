import React from 'react';

export const FilterDropDown = (props) => (
  <select onChange={ event => props.onChange(event.target.value) } >
    <option key='all' value=''>Any { props.name }</option>
    { props.options.map(
      t => <option
        key={ t }
        value={ t }>
        { t }</option>
    ) }
  </select>
);
