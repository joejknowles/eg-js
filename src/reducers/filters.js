import { combineReducers } from 'redux';

export const typeReducer = (state = '', action) => {
  if (action.type === 'SET_TYPE_FILTER') {
    return action.filter;
  }
  return state
};

export default combineReducers({ type: typeReducer });
