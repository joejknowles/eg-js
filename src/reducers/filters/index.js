import { combineReducers } from 'redux';

import options from './options'

export const typeReducer = (state = '', action) => {
  if (action.type === 'SET_TYPE_FILTER') {
    return action.filter;
  }
  return state
};

export const location = (state = '', action) => {
  if (action.type === 'SET_LOCATION_FILTER') {
    return action.filter;
  }
  return state
};

export const titleSearch = (state = '', action) => {
  if (action.type === 'SET_TITLE_SEARCH') {
    return action.term;
  }
  return state
};

export default combineReducers({ type: typeReducer, location, titleSearch, options });
