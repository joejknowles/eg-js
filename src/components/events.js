import React, { Component } from 'react';
import { connect } from 'react-redux';

import Listing from './listing';
import LoadingIndicator from './loadingIndicator';
import { eventsSelector, eventsLoadingSelector } from '../reducers'

export class Events extends Component {
  componentDidMount() {
    this.props.fetchEvents();
  }

  render() {
    const { props } = this;
    if (props.eventsLoading) return (<LoadingIndicator />);
    return (
      <div className="events-list">
        {
          props.events.map(
            id => <Listing id={ id } key={ id } />
          )
        }
      </div>
    );
  }
}

const mapStateToProps = (state) => ({
  events: eventsSelector(state),
  eventsLoading: eventsLoadingSelector(state)
});

const mapDispatchToProps = (dispatch) => ({
  fetchEvents: () => dispatch({ type: "FETCH_EVENTS_REQUEST" })
});

export default connect(mapStateToProps, mapDispatchToProps)(Events);
