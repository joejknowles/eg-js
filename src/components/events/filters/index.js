import React from 'react';
import { connect } from 'react-redux';

import { FilterDropDown } from './dropDown'
import {
  setTypeFilter,
  setLocationFilter,
  setTitleSearch
} from '../../../actions';

import { typeFilter, locationFilter, titleSearch, typeOptions, locationOptions } from '../../../reducers';

export const Filters = (props) => (
  <div className='Filters'>
    <FilterDropDown selected={ props.type }
      name="type"
      options={ props.typeOptions }
      onChange={ props.onTypeFilterChange }/>
    <FilterDropDown selected={ props.location }
      name="location"
      options={ props.locationOptions }
      onChange={ props.onLocationFilterChange }/>
    <input placeholder="search..." value={ props.titleSearch }
      onChange={ (e) =>  props.onTitleSearchChange(e.target.value) }
    />
  </div>
);

const mapStateToProps = (state) => ({
  type: typeFilter(state),
  location: locationFilter(state),
  titleSearch: titleSearch(state),
  typeOptions: typeOptions(state),
  locationOptions: locationOptions(state)
});

const mapStateToDispatch = (dispatch) => {
  return {
    onTypeFilterChange: (filter) => dispatch(setTypeFilter(filter)),
    onLocationFilterChange: (filter) => dispatch(setLocationFilter(filter)),
    onTitleSearchChange: (term) => dispatch(setTitleSearch(term))
  }
};

export default connect(mapStateToProps, mapStateToDispatch)(Filters);
