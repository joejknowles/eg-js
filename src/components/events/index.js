import React, { Component } from 'react';
import { connect } from 'react-redux';

import Listing from '../listing';
import LoadingIndicator from '../loadingIndicator';
import { eventsFirstLoadingSelector } from '../../reducers';
import { filteredEventsSelector } from '../../selectors/filter'

export class Events extends Component {
  componentDidMount() {
    this.props.fetchEvents();
  }

  render() {
    const { props } = this;
    if (props.eventsLoading) return (<LoadingIndicator />);
    return (
      <div className="events-list">
        {
          props.events.map(
            id => <Listing id={ id } key={ id } />
          )
        }
      </div>
    );
  }
}

const mapStateToProps = (state) => ({
  events: filteredEventsSelector(state),
  eventsLoading: eventsFirstLoadingSelector(state)
});

const mapDispatchToProps = (dispatch) => ({
  fetchEvents: () => dispatch({ type: "FETCH_EVENTS_REQUEST" })
});

export default connect(mapStateToProps, mapDispatchToProps)(Events);
