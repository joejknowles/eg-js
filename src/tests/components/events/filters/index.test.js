import { Filters } from '../../../../components/events/filters';

import React from 'react';
import ReactTestUtils from 'react-addons-test-utils';
import { mount } from 'enzyme';

const shallowRenderer = ReactTestUtils.createRenderer();

describe('Filters component', () => {
  it('renders as before', () => {
    const component = shallowRenderer.render(<Filters />);
    expect(component).toMatchSnapshot();
  });

  it('calls on type filter Change', () => {
    const mock = jest.fn();
    const component = mount(
      <Filters
        onTypeFilterChange={ mock }
      />
    );
    component.find('select').first().simulate('change', { target: { value: 'gigs' } });
    expect(mock).toHaveBeenCalledWith('gigs');
  });
});
