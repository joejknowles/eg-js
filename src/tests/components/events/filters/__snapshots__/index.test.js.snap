exports[`Filters component renders as before 1`] = `
<div
  className="Filters">
  <Unknown
    name="type"
    onChange={undefined}
    options={undefined}
    selected={undefined} />
  <Unknown
    name="location"
    onChange={undefined}
    options={undefined}
    selected={undefined} />
  <input
    onChange={[Function]}
    placeholder="search..."
    value={undefined} />
</div>
`;
